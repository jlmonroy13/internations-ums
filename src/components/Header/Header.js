import React from 'react';
import PropTypes from 'prop-types';

function Header({ setNavState, isNavOpen }) {
  function onSetNavState() {
    setNavState(!isNavOpen);
  }
  return (
    <header className="header">
      <h1 className="header__title">
        <span className="icon icon--logo" /> InterNations{' '}
        <span className="header__title--pink">UMS</span>
      </h1>
      <span className="icon icon--menu header__btn" onClick={onSetNavState}/>
    </header>
  );
}

Header.propTypes = {
  setNavState: PropTypes.func,
  isNavOpen: PropTypes.bool,
};

export default Header;
