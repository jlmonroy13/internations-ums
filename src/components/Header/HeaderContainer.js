import { connect } from 'react-redux';
import { setNavState } from '../../actions/events';
import Header from './Header';

function mapStateToProps(state) {
  const { events: { isNavOpen } } = state;
  return { isNavOpen };
}

function mapDispatchToProps(dispatch) {
  return {
    setNavState: state => dispatch(setNavState(state)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
