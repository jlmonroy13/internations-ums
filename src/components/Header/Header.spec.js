import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('<Account />', () => {
  it('should render the header component', () => {
    const wrapper = renderer.create(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
