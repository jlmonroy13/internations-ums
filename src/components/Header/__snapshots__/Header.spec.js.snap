// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> should render the header component 1`] = `
<header
  className="header"
>
  <h1
    className="header__title"
  >
    <span
      className="icon icon--logo"
    />
     InterNations
     
    <span
      className="header__title--pink"
    >
      UMS
    </span>
  </h1>
  <span
    className="icon icon--menu header__btn"
    onClick={[Function]}
  />
</header>
`;
